File: 052_Quiz - inheritance/src/hr/unipu/java/EShortAnswerQuestion.java
package hr.unipu.java;

// Najava implementacije sučelja
public class EShortAnswerQuestion implements Question  {
    String text;
    String correctAnswer;   // proširena dostupnost na paketno-privatnu (-> obrisati private)


    public EShortAnswerQuestion(String text, String correctAnswer) {
        this.text = text;
        this.correctAnswer = correctAnswer;
    }

    public String getText() {
        return text;
    }

    public boolean isCorrectAnswer(String answer) {
        return correctAnswer.equalsIgnoreCase(answer);
    }

}

File: 052_Quiz - inheritance/src/hr/unipu/kotlin/EShortAnswerQuestion.kt
package hr.unipu.kotlin

// Najava implementacije sučelja
// - proširena dostupnost na paketno-privatnu (-> obrisati private)
open class EShortAnswerQuestion(text: String,
                                internal var correctAnswer: String) : Question {
    override lateinit var text: String
        internal set


    init {
        this.text = text
    }

    override fun isCorrectAnswer(answer: String): Boolean {
        return correctAnswer.equals(answer, ignoreCase = true)
    }

}
